### ISSUE TITLE
Deep-object-reid training performance

### ISSUE THREAD

**Issue by ChienHsuan** [openvinotoolkit/training_extensions#593]
I recently tried the deep-object-reid project, but I couldn't get the same result described in paper. I am not sure what the problem is. The following are some settings and references. @sovrasov 

+ Paper: https://arxiv.org/abs/2003.07618
+ Environment:
  Python 3.7.11
  Pytorch 1.8.1 LTS
  Opencv 4.4.0-openvino
+ Dataset: I got MSMT17 dataset from https://github.com/alibaba/cluster-contrast-reid
+ Config: I deleted some unsupported options from **OSNet_IAP_0_75x_msmt_all.yaml**.
```
model:
  name: 'fpn_osnet_x0_75'
  pretrained: True
  feature_dim: 256
  IN_first: True
  pooling_type: 'conv'
  fpn:
    enable: False

data:
  root: './'
  sources: ['msmt17']
  targets: ['market1501']
  height: 256
  width: 128
  combineall: True
  save_dir: 'outputs/reid-exp'
  transforms:
    random_flip:
      enable: True
      p: 0.5
    random_erase:
      enable: True
      p: 0.5
    random_figures:
      enable: True
      p: 0.33
    random_padding:
      enable: True
      p: 0.25
    color_jitter:
      enable: True
      p: 0.8
      saturation: 0.1
      hue: 0.1
      brightness: 0.15
      contrast: 0.15
    random_rotate:
      enable: True
      p: 0.33
    random_gray_scale:
      enable: True
      p: 0.1
    random_grid:
      enable: True
      p: 0.15

loss:
  name: 'am_softmax'
  softmax:
    label_smooth: False
    conf_penalty: 0.3
    m: 0.35
    s: 30.0
    pr_product: True

sampler:
  train_sampler: 'RandomIdentitySamplerV2'

metric_losses:
  enable: False

train:
  optim: 'amsgrad'
  lr: 0.0015
  max_epoch: 65
  batch_size: 128
  fixbase_epoch: 5
  open_layers: ['classifier', 'fc', 'global_avgpool']
  lr_scheduler: 'multi_step'
  stepsize: [40, 50]
  warmup: 0

test:
  batch_size: 300
  dist_metric: 'cosine'
  normalize_feature: True
  evaluate: False
  eval_freq: 10
  rerank: False
  visactmap: False
```
+ Running command: bash train.sh
```
DATA_DIR="/home/user"
CONFIG_FILE="configs/OSNet_IAP_0_75x_msmt_all.yaml"

export CUDA_VISIBLE_DEVICES=5,6

python tools/main.py \
--config-file ${CONFIG_FILE} \
--root ${DATA_DIR} \
--gpu-num 2
```
+ Result: **OSNet-IAP 0.75x, MSMT17-all** 52.49mAP in paper.
  But I just got 5.6% mAP.
```
2021-08-16_02:15:10|##### Evaluating market1501 (target) #####
2021-08-16_02:15:30|** Results (main_model) **
2021-08-16_02:15:30|mAP: 5.60%
2021-08-16_02:15:30|CMC curve
2021-08-16_02:15:30|Rank-1  : 11.64%
2021-08-16_02:15:30|Rank-5  : 28.53%
2021-08-16_02:15:30|Rank-10 : 39.16%
2021-08-16_02:15:30|Rank-20 : 50.65%
2021-08-16_02:15:30|Checkpoint saved to "outputs/reid-exp/main_model/main_model.pth.tar-65"
```

**Comment by sovrasov** [comment]
We are currently using deep-object-reid repo for image classification and reid branches of code could be broken here.
I guess 5.60 mAP means that the model can't even converge on MSMT17 (though I haven't seen the loss curve).
As a quick fix I suggest picking cf14ca77e780a15f6ce9946d8af8a856594ee9d9 in deep-object-reid as a commit which is the closest to the last update of configs here.

**Comment by ChienHsuan** [comment]
Thanks for your quick reply. Could I ask which version of *deep-object-reid* can reach the performance mentioned in the paper? I have used the same settings(AM-Softmax, sampler, model architecture, etc.) in original *deep-person-reid*, but I got the much lower cross-domain accuracy. So I would like to know how you solve these problem.

**Comment by sovrasov** [comment]
I've mentioned the commit in deep-object-reid that could reproduce results from the paper on MSMT17. deep-person-reid has lower cross-domain performance on this dataset since the author didn't consider this scenario. Which scenario are you interested in? MSMT->Market or something else?

We don't support algorithms from the misc folder anymore, so reid is provided as is now. I'll look at the current state of deep-object-reid and check if the reid part works, when I have a time.

**Comment by ChienHsuan** [comment]
Ok, thanks for your help.
